packages/snowpack/src/index.js

```diff
@@ -2,15 +2,16 @@ import { compareSignatures, isComponent } from '@prefresh/utils';
 
 export default function preactRefreshPlugin(config, pluginOptions) {
 	return {
-		knownEntrypoints: ['@prefresh/core'],
-		async transform({ contents, urlPath, isDev }) {
-			if (!isDev || !urlPath.endsWith('.js') || config.devOptions.hmr === false)
-				return;
+		name: '@prefresh/snowpack',
+		input: ['.js'],
+		output: ['.js'],
+		async build({ contents, filePath, isDev }) {
+			if (!isDev || config.devOptions.hmr === false) return contents;
 
 			return {
 				result: `
           import '@prefresh/core';
-          import * as $OriginalModule$ from ${JSON.stringify(urlPath)};
+          import * as $OriginalModule$ from ${JSON.stringify(filePath)};
```

Comment:
This is the only change that I have a question on. In our new plugin system, we’d like to relieve plugins from the responsibility from dev vs build plugins. So build plugins are engineered toward build-time transformations, and we’ve removed some dev hooks for simplicity.

Moving forward, I don’t think we’ll give plugins the ability to see what URL is being requested from the dev server (it’s not deterministic, and we’d like to not provide an environment where the contents of files change based on URLs requested, etc.).

That said, it seems this plugin operates off the original file path. However it… requests itself? I’d like to think through more of what this plugin is doing to get around the limitation of it no longer being able to see request URLs.

Reply:
This was needed to obtain a reference to the oldModule since this was available in the byteCache, this could be replaced with something like https://github.com/JoviDeCroock/prefresh/blob/main/packages/vite/src/index.js#L58-L70